### ISSUE TITLE
How to convert the. bag packet of ROS to the. record packet of Cyber

### ISSUE THREAD

**Issue by tjc12581** [ApolloAuto/apollo#15269]
How to convert the. bag packet of ROS to the. record packet of Cyber

**Comment by daohu527** [comment]
Yes, We already have a tool [bag_convert](https://github.com/daohu527/bag_convert), which currently only converts a limited type of messages(lidar, imu, pose).

There are currently two solutions. One is to support ros messages to Apollo on a 1:1 basis. The other is to directly convert ros messages into Apollo format.
The former is more general and the latter is more convenient, so I don't know if I need to support both

**Comment by tjc12581** [comment]
I recorded a data packet using ROS and I want to change the communication framework to cyberRT. I need to convert the. bag data packet to. record format and ensure that the content inside the packet does not change. Can you tell me the specific command or method

**Comment by daohu527** [comment]
> Yes, We already have a tool [bag_convert](https://github.com/daohu527/bag_convert), which currently only converts a limited type of messages(lidar, imu, pose).
> 
> There are currently two solutions. One is to support ros messages to Apollo on a 1:1 basis. The other is to directly convert ros messages into Apollo format. The former is more general and the latter is more convenient, so I don't know if I need to support both

You can try the above tools to see if they can meet your requirements
